runbotics/runbotics-scheduler/src/scheduler-database/notification-bot/notification-bot.controller.ts

```diff
@@ -0,0 +1,50 @@
+import { Body, Controller, Delete, Get, HttpCode, HttpStatus, Logger, Param, ParseIntPipe, ParseUUIDPipe, Post, UseInterceptors } from '@nestjs/common';
+
+import { TenantInterceptor } from '#/utils/interceptors/tenant.interceptor';
+import { ZodValidationPipe } from '#/utils/pipes/zod-validation.pipe';
+import { User } from '#/utils/decorators/user.decorator';
+import { UserEntity } from '#/database/user/user.entity';
+
+import { createNotificationBotSchema, CreateNotificationBotDto } from './dto/create-notification-bot.dto';
+import { NotificationBotService } from './notification-bot.service';
+
+
+
+@UseInterceptors(TenantInterceptor)
+@Controller('/api/scheduler/tenants/:tenantId/notifications-bot')
+export class NotificationBotController {
+    private readonly logger = new Logger(NotificationBotController.name);
+
+    constructor(
+        private readonly notificationBotService: NotificationBotService
+    ) {}
+
+    @Get('bots/:botId')
+    getAllNotificationsByBotId(
+        @Param('botId', ParseIntPipe) botId: number,
+        @User() user: UserEntity,
+    ) {
+        this.logger.log('REST request to get all bot notifications by bot id: ', botId);
+        return this.notificationBotService.getAllByBotId(botId, user);
+    }
+
+    @Post()
+    createNotification(
+        @Body(new ZodValidationPipe(createNotificationBotSchema))
+        notificationBotDto: CreateNotificationBotDto,
+        @User() user: UserEntity
+    ) {
+        this.logger.log(`REST request to create bot notification by user id: ${user.id}`);
+        return this.notificationBotService.create(notificationBotDto, user);
+    }
+
+    @Delete(':id')
+    @HttpCode(HttpStatus.NO_CONTENT)
+    async deleteNotification(
+        @Param('id', ParseUUIDPipe) id: string,
+        @User() user: UserEntity,
+    ) {
+        this.logger.log('REST request to delete bot notification by id: ', id);
+        await this.notificationBotService.delete(id, user);
```

Comment:
nvm, I was wrong

Reply:
Are you sure ? If we return promise then NestJS will resolve it before returning http response. But in this case how does it know if we didn't return after delete method? There could be an error but Nest will return 204... I've tested it and it would be a critical app crash.